ibis/pandas/execution/window.py

```diff
@@ -274,18 +281,20 @@ def execute_window_op(
     else:
         if order_by:
             source, grouping_keys, ordering_keys = util.compute_sorted_frame(
-                data, order_by, **kwargs
+                data, order_by, timecontext=new_timecontext, **kwargs
             )
             post_process = _post_process_order_by
         else:
             source = data
             post_process = _post_process_empty
 
-    new_scope = toolz.merge(
-        scope,
-        OrderedDict((t, source) for t in operand.op().root_tables()),
-        factory=OrderedDict,
-    )
+    additional_scope = {}
+    for t in operand.op().root_tables():
```

Comment:
Why changing the list comprehension to for loop?

Reply:
There might be better ways but let me explain why, scope_item() now return with a dict {op: {"value":xx, "timecontext":xx}},  how can I turn it into (key, value) like (t, source) in a one-linere we had previously?

Reply:
I think your code is the same as 

```
toolz.merge(
    set_scope_item(t, source, adjusted_timecontext)
    for t in
    operand.op().root_tables()
)
```
?

Reply:
+1